webapp/src/main/java/com/box/l10n/mojito/service/asset/VirtualTextUnitBatchUpdaterService.java

```diff
@@ -313,4 +325,16 @@ void createMappedAssetTextUnit(
     assetTextUnitToTMTextUnit.setTmTextUnit(tmTextUnitRepository.getOne(tmTextUnitId));
     assetTextUnitToTMTextUnitRepository.save(assetTextUnitToTMTextUnit);
   }
+
+  @Async
+  void scheduleAITranslation(Set<TMTextUnit> textUnits, Repository repository) {
+    for (TMTextUnit textUnit : textUnits) {
+      logger.debug(
+          "Sending tmTextUnitId: {} information to AI Translation scheduler", textUnit.getId());
+      AITranslateJobInput aiTranslateJobInput = new AITranslateJobInput();
```

Comment:
Would you expect the system to perform Ok with this level of granularity for jobs scheduled? (vs. having these jobs with a repo + locale + defined batch size of TU ids for example).

Don't have a predefined opinion on this needing change - just something we should consider re impact on Quartz processing.

Reply:
This might also impact our ability to effectively do client-side rate-limiting (vs. in practice exhausting a lot/all of the retries), depending on the number of parallel AITranslateJobInput jobs that would execute concurrently.

Reply:
Good point, I think that as this flow is only triggered off new text units and our present rate of new text units is relatively low I don't see a performance problem at present. However if we were to get a number of text units received at the same time it's possible we hit our rate-limit and then the backoffs are not sufficient to successfully translate text units to all locales.

In that scenario a buffered job may be more useful to restrict the number of requests we are potentially making in parallel e.g. instead of a one to one mapping between text unit and job we potentially buffer the scheduling and pass a list of text units to the `AITranslateJob`, and the job itself works through the list so we reduce the potential parallelization. Trade off here is higher reliability vs lower speed, picking the granularity at which we buffer at also impacts it e.g. 1 job executing for all repos vs 1 job per repo (increasing the risk of encountering rate limit issues)